### PR TITLE
Issue #12017: Kill surviving mutations in AbstractFileSetCheck related to fileExtensions

### PR DESCRIPTION
**Pull request by Vyom-Yadav**
#12017

### This mutation falls under the category:

- Logic was analyzed to develop the test case.

### CODE REVIEW

File: src/test/java/com/puppycrawl/tools/checkstyle/api/AbstractFileSetCheckTest.java [hunk 1]
```diff
@@ -263,6 +264,20 @@ public void testMultiFileFireErrors() throws Exception {
                 .isEqualTo(2);
     }
 
+    /**
+     * S2384 - Mutable members should not be stored or returned directly.
```
Comment (nick-mancuso) [inline src/test/java/com/puppycrawl/tools/checkstyle/api/AbstractFileSetCheckTest.java:268]: Please expand comment similar to https://github.com/checkstyle/checkstyle/pull/12023#discussion_r938703126

**Review by nick-mancuso — CHANGES_REQUESTED** [review]
@Vyom-Yadav please resolve conflicts 

Item:

File: src/test/java/com/puppycrawl/tools/checkstyle/api/AbstractFileSetCheckTest.java (hunk 1 above)
Comment (Vyom-Yadav) [reply src/test/java/com/puppycrawl/tools/checkstyle/api/AbstractFileSetCheckTest.java:268]: Done.

File: src/test/java/com/puppycrawl/tools/checkstyle/api/AbstractFileSetCheckTest.java [hunk 2]
```diff
@@ -263,6 +264,22 @@ public void testMultiFileFireErrors() throws Exception {
...
+    @Test
+    public void testCopiedArrayIsReturned() {
+        final DummyFileSetCheck check = new DummyFileSetCheck();
+        final String[] extensions = {".tmp"};
+        check.setFileExtensions(extensions);
+        final String[] checkExtensions = TestUtil.getInternalState(check, "fileExtensions");
+        assertWithMessage("Extensions should be copied")
+                .that(checkExtensions)
+                .isNotSameInstanceAs(check.getFileExtensions());
+    }
```
Comment (baratali) [inline src/test/java/com/puppycrawl/tools/checkstyle/api/AbstractFileSetCheckTest.java:281]: we can avoid reflection here. 
also, the setter supports varargs, so the code can be simplified
```java
@Test
public void testCopiedArrayIsReturned() {
    final DummyFileSetCheck check = new DummyFileSetCheck();
    check.setFileExtensions(".tmp");
    assertWithMessage("Extensions should be copied")
            .that(check.getFileExtensions())
            .isNotSameInstanceAs(check.getFileExtensions());
}
```

Comment (Vyom-Yadav) [reply src/test/java/com/puppycrawl/tools/checkstyle/api/AbstractFileSetCheckTest.java]: Good suggestion, changes done.

File: src/test/java/com/puppycrawl/tools/checkstyle/api/AbstractFileSetCheckTest.java [hunk 3]
```diff
@@ -34,6 +34,7 @@
 import com.puppycrawl.tools.checkstyle.AbstractModuleTestSupport;
 import com.puppycrawl.tools.checkstyle.Checker;
 import com.puppycrawl.tools.checkstyle.DefaultConfiguration;
+import com.puppycrawl.tools.checkstyle.internal.utils.TestUtil;
```
Comment (baratali) [inline src/test/java/com/puppycrawl/tools/checkstyle/api/AbstractFileSetCheckTest.java:37]: not needed anymore

**Review by baratali — CHANGES_REQUESTED** [review]

File: src/test/java/com/puppycrawl/tools/checkstyle/api/AbstractFileSetCheckTest.java (hunk 3 above)
Comment (Vyom-Yadav) [reply src/test/java/com/puppycrawl/tools/checkstyle/api/AbstractFileSetCheckTest.java]: Done.

**Review by baratali — APPROVED** [review]

**Review by nick-mancuso — APPROVED** [review]

**Comment by nick-mancuso** [comment]
TC is failing in all projects right now, it is not related to changes here.

**Pull request merged**